### DISCUSSION TITLE
TextBundle examples.

### DISCUSSION THREAD

**Discussion by The-DevBlog** [bevyengine/bevy#2585] [Q&A]
Is there a way to use the 2d TextBundle with using `App::build()` instead of `App::new()`? I currently am using Bevy's main release branch and not wanting to port things over to another pre-release version at the moment. I cloned down the [text2d example](https://github.com/bevyengine/bevy/blob/main/examples/2d/text2d.rs) and it works fine. When I change the code from `App::new()` to `App::build()` and add the `.system()` to my systems, I can compile but nothing renders on the game window. 

[Link to repo](https://github.com/AndrewCS149/Blazing-Bevy/tree/playInstructions/Blazing-Bevy/platformer)

App::new()
```rust
fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_startup_system(setup)
        .add_system(animate_translation)
        .add_system(animate_rotation)
        .add_system(animate_scale)
        .run();
}
```

App::build()
```rust 
    App::build()
        .add_plugins(DefaultPlugins)
        .add_startup_system(setup.system())
        .add_system(animate_translation.system())
        .add_system(animate_rotation.system())
        .add_system(animate_scale.system())
        .run();
```

**Comment by IceSentry** [comment]
Have you seen this one?

https://github.com/bevyengine/bevy/blob/main/examples/ui/text.rs

If there's something you don't understand in the example, feel free to ask follow up questions here or on discord

**Reply by The-DevBlog** [reply]
I did see this, but I dont understand one thing. I can clone this down to my machine and run it fine. But when I try to port it over to my own project, the text does not render. The only real difference I see is the `App::new()` in the example you posted and the `App::build()` I am using in my own project. What is the difference?

Example:

App::new
```rust
fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_plugin(FrameTimeDiagnosticsPlugin::default())
        .add_startup_system(setup)
        .add_system(text_update_system)
        .add_system(text_color_system)
        .run();
}
```

App::build
```rust
fn main() {
    App::build()
        .add_plugins(DefaultPlugins)
        .add_plugin(FrameTimeDiagnosticsPlugin::default())
        .add_startup_system(setup)
        .add_system(text_update_system)
        .add_system(text_color_system)
        .run(); 
}
```

**Reply by IceSentry** [reply]
Oh, yeah, that's a new feature in main that isn't released yet. I should have linked you to the currently released branch. Unless you are already using main, then in that case it's just a matter of merging App and AppBuilder and just have App do everything. 

If you have a way to reproduce your situation you can share it here and maybe someone will see what the problem is.

**Reply by The-DevBlog** [reply]
Awesome that clarifies a lot @IceSentry. I just updated my question to reflect what I'm trying to do. From what I see there, the main release doesn't support the 2D text bundle? That being said it compiles just fine, but no text is rendering.

**Reply by IceSentry** [reply]
So to be clear, there's bevy 0.5.0 which is the latest release on crates.io and there is bevy main which is obtained by pointing your Cargo.toml to this repository with the main branch and that will contain all the most recent changes like not needing `App::build()` and `.system()`.

With that in mind, when you say that the example works but your code doesn't, which version of bevy are you using for your own code and which version are you using for the example? The change to the app builder and systems should have no impact on rendering text and the text example should still work with both versions.

Also, is your project in a public repository? If you not, can you make a small public repo or even just a code snippet that shows the issue you are having? The app builder alone isn't enough to diagnose what is going on without seeing some code and it shouldn't be causing text to not render.

**Reply by IceSentry** [reply]
Wait, so reading your updated post again, do you just mean you changed the example code you cloned to use .system() and App::build and it compiled but nothing rendered? Because if you did that on main it shouldn't even compile since App::build() doesn't exist anymore

**Reply by The-DevBlog** [reply]
I just updated my discussion post with the link to the repo ^

The only dependency I'm using is `bevy = "0.5.0"`. So looks like I've been using the regular crates.io one. How would I point the .toml file to this repo? 

I was incorrect what I said earlier. I can't simply change `App::build()` and `App::new()`. I can only use `App::build()` because of the crates.io dependency. 

Also, thanks so much for all this info! It's really clearing the air on a lot of things.

**Reply by IceSentry** [reply]
Here's the cargo book section about git dependencies https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories. For the actual problem I'll have a look at the repo later

**Reply by IceSentry** [reply]
So I did a quick look around of your repo and I'm not seeing any textbundle. Where exactly are you inserting it?

**Reply by The-DevBlog** [reply]
My apologies. I pointed you to the wrong branch. [Here it is](https://github.com/AndrewCS149/Blazing-Bevy/tree/playInstructions/Blazing-Bevy/platformer).

The Text2dBundle is located in the `setup` function in `main.rs`. It's just some simple code to try to get some text on the window.
